docs/guidelines/platforms/android/guideline_operable_android.md

```diff
@@ -146,63 +146,37 @@ Apps should not contain elements that flash more than three times in one second.
 
-_Note 1. Navigation mode can be chosen by swiping **up** and **down** when using **TalkBack**. Once Headers is chosen as a Navigation option, the user can navigate through **headers** by swiping right and left instead of navigating through single items.._
+Once that is done, it will get picked up by TalkBack and the user will be able to skip parts of content by navigating through headings. This should be automatically satisfied by following the [Heading and Labels guideline](guideline_operable_android.md#heading-and-labels-wcag-246---level-aa).
 
-```
-<TextView
-    android:id="@+id/personalData" ...
-    android:text="@string/personal_data_heading"
-    android:accessibilityHeading="true"/>
+#### Controls
+TalkBack can recognize that something is a control based on the element's role (button, toggle, edit text etc.). Native controls will always be recognized automatically. For custom components, make sure to set a proper role as described in the [Name, Role, Value chapter](guideline_robust_android.md#name-role-value-wcag-412---level-a).
 
-<EditText
-    android:id="@+id/nameEntry" ... />
+#### Links
+This criteria should be satisfied by following the recommendations from [Link purpose guideline](guideline_operable_android.md#link-purpose).
 
-<EditText
-    android:id="@+id/surnameEntry" ... />
-
-<TextView
-    android:id="@+id/workplaceData" ...
-    android:text="@string/personal_data_heading"
-    android:accessibilityHeading="true"/>
+#### Skippable groups
 
-<EditText
-    android:id="@+id/addressEntry" ... />
-```
+In addition to the above, a section of a screen containing numerous items should have related items grouped, so that they are easily skippable and do not require multiple swipes to go over. This feature is based on a good implementation of grouping as described in [Info and relationships - Element relationships guideline](guideline_percievable_android.md#element-relationships).
 
-For apps with minSdk < 28, headings can be defined programmatically using ViewCompat.
-
-An example is given down below:
-
-```
-ViewCompat.setAccessibilityDelegate(personalData, object : AccessibilityDelegateCompat() {
-    override fun onInitializeAccessibilityNodeInfo(host: View?, info: AccessibilityNodeInfoCompat?) {
-        super.onInitializeAccessibilityNodeInfo(host, info)
-        info?.isHeading = true
-    }
-})
-```
-
-:no_entry_sign: **Failure criteria**
+🚫 **Failure criteria**
 
 - The user of accessibility services has to navigate through all the items displayed on the screen with no possibility to fasten the navigation process.
+    - No elements set as headings to separate bigger parts of text or groups in general.
+    - Controls not being recognized as such. This can happen when, for example, using `TextView` instead of a `Button` or `ImageView` instead of a `Checkbox` or a `Switch` without any accessibility info modifications.
```

Comment:
I see that you reference mainly views here, how does this apply to compose? Do we have a similar issue there?

Reply:
I actually wasn't able to reproduce it with Compose, I guess it works somehow differently, so all clickable elements are picked up. 🤔